lib/sunrise/sunrise.php

```diff
@@ -41,10 +47,38 @@ function site_not_found( $network, $domain, $path ) {
 	];
 	$data = json_encode( $data );
 
-	trigger_error( 'Site Not Found! ' . $data, E_USER_WARNING );
+	trigger_error( 'ms_site_not_found: ' . $data, E_USER_WARNING );
+
+	handle_not_found_error( 'site' );
 }
 add_action( 'ms_site_not_found', __NAMESPACE__ . '\site_not_found', 9, 3 ); // Priority 9 to log before WP_CLI kills execution
 
+function handle_not_found_error( $error_type ) {
+	$is_healthcheck = Context::is_healthcheck();
+	if ( $is_healthcheck ) {
+		http_response_code( 200 );
+		header( 'Content-type: text/plain' );
+		echo sprintf( '%s not found; but still OK', esc_html( $error_type ) );
+		exit;
+	}
+
+	$is_web_request = Context::is_web_request();
+	if ( $is_web_request ) {
+		$is_maintenance_mode = Context::is_maintenance_mode();
+		if ( $is_maintenance_mode ) {
+			$status_code = 200;
```

Comment:
200 will cause this to be cached. Is that ideal? If we send 503 here, will it trigger health checks?

https://yoast.com/http-503-site-maintenance-seo/

Reply:
Ah, good catch. We're setting a `503` in `000-vip-init.php` as well.

Reply:
> If we send 503 here, will it trigger health checks?

The healthcheck handler fires earlier, so shouldn't be an issue.